test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+            });
+            expect(array.length).to.eql(5);
+            expect(array).to.include(1);
+            expect(array).to.include(5);
+            expect(_.reduce.apply(_, datasets.reduce())).to.eql(55);
+        });
+    });
+
+    describe('sortBy', function () {
+        var params = datasets.sortBy();
+        var array = params[0];
+
+        it('should match types of passing elements', function () {
+            var element1 = params[0] instanceof Array;
+            var element2 = typeof params[1];
+
+            expect(element1).to.eql(true);
+            expect(element2).to.eql('string');
+            expect(params.length).to.eql(2);
+        });
+        it('should contain array of random strings with different lenghts', function () {
+            var checkLetters = array[0].substr(0, 3);
+            var checkLetters2 = array[2].substr(2, 5);
+            var checkLetters3 = array[3].substr(6, 10);
+            var checkLetters4 = array[4].substr(3, 6);
+
+            expect(array).to.have.length(6);
+            array.forEach(function (item, index) {
+                expect(typeof item).to.eql('string');
+                if (index == 0) {
+                    expect(item).to.have.length(4);
+                    expect(item).to.not.contain(checkLetters2);
+                    expect(item).to.not.contain(checkLetters3);
+                    expect(item).to.not.contain(checkLetters4);
+                } else if (index == 1) {
+                    expect(item).to.have.length(2);
+                } else if (index == 2) {
+                    expect(item).to.have.length(6);
+                    expect(item).to.not.contain(checkLetters);
+                    expect(item).to.not.contain(checkLetters3);
+                    expect(item).to.not.contain(checkLetters4);
+                } else if (index == 3) {
+                    expect(item).to.have.length(10);
+                    expect(item).to.not.contain(checkLetters);
+                    expect(item).to.not.contain(checkLetters2);
+                    expect(item).to.not.contain(checkLetters4);
+                } else if (index == 4) {
+                    expect(item).to.have.length(8);
+                    expect(item).to.not.contain(checkLetters);
+                    expect(item).to.not.contain(checkLetters2);
+                    expect(item).to.not.contain(checkLetters3);
+                } else if (index == 5) {
+                    expect(item).to.have.length(3);
+                }
+            })
+        });
+        it('should return sorted array by length of each string', function () {
+            var item1 = array[0];
+            var item2 = array[1];
+            var item3 = array[2];
+            var item4 = array[3];
+            var item5 = array[4];
+            var item6 = array[5];
+
+            expect(_.sortBy.apply(_, datasets.sortBy())).to.eql([item2, item6, item1, item3, item5, item4]);
+        });
```

Comment:
What about other types??? (Array|Array[]|Function|Function[]|Object|Object[]|string|string[])
This is barely 10% of the test cases for this exercise and to make it worst it is poor I think. Not to mention possible object collections.